src/Stan/Pattern/Ast.hs

```diff
@@ -193,3 +220,25 @@ tuple =
     PatternAstNode (one ("HsTupleTy", "HsType"))  -- tuple type
     |||
     PatternAstNode (one ("ExplicitTuple", "HsExpr"))  -- tuple literal
+
+{- | Pattern for a single @guard@ branch:
+
+@
+    | x < y = ...
+@
+-}
+guardBranch :: PatternAst
+guardBranch = PatternAstNode $ one ("BodyStmt", "StmtLR")
+
+{- | Pattern for the right-hand-side. Usually an equality sign.
+
+@
+   foo = baz
+@
+-}
+rhs :: PatternAst
+rhs = PatternAstNode $ one ("GRHS", "GRHS")
```

Comment:
Oh, that what it means! I used the same and called it `patternMatchingArrow` assuming that it would be used only there. I can reuse it!